extensions/control-plane/api/management-api/asset-api/src/main/java/org/eclipse/edc/connector/api/management/asset/model/AssetCreationRequestDto.java

```diff
@@ -32,9 +32,21 @@ private AssetCreationRequestDto() {
     }
 
     @JsonIgnore
-    @AssertTrue(message = "no empty property keys")
+    @AssertTrue(message = "no empty property keys and no duplicate keys")
     public boolean isValid() {
-        return properties != null && properties.keySet().stream().noneMatch(it -> it == null || it.isBlank());
+        //check for empty keys
+        boolean validPrivate = privateProperties != null && privateProperties.keySet().stream().noneMatch(it -> it == null || it.isBlank());
```

Comment:
what i meant previously was to extract the check into a separate method to avoid code duplication. While that may seem a bit over-the-top, it will centralize a check, that applies to both maps, in one single place, which makes maintenance easier and reduces the chance for errors.

Reply:
moved the checks from `AssetUpdateRequestDto` and `AssetCreationRequestDto` into `AssetRequestDto`.
there is now only a call to these methods left in the original files.